enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk_setup/analytics.ts

```diff
@@ -5,66 +5,74 @@ import type {
 } from "metabase-enterprise/embedding_iframe_sdk/types/embed";
 
 import type { SdkIframeEmbedSetupExperience } from "./types";
-import { filterEmptySettings } from "./utils/filter-empty-settings";
 
-export const trackEmbedWizardExperienceSelected = (
+/**
+ * Tracking every embed options would be too much, so we only track
+ * the most relevant options that reveal usage patterns.
+ */
+const EMBED_OPTIONS_TO_TRACK = [
+  "drills",
+  "withTitle",
+  "withDownloads",
+  "isSaveEnabled",
+  "readOnly",
+] as const satisfies SdkIframeEmbedSettingKey[];
+
+export const trackEmbedWizardOpened = () =>
+  trackSimpleEvent({ event: "embed_wizard_opened" });
+
+export const trackEmbedWizardExperienceCompleted = (
   experience: SdkIframeEmbedSetupExperience,
 ) =>
   trackSimpleEvent({
-    event: "embed_wizard_experience_selected",
+    event: "embed_wizard_experience_completed",
     event_detail: experience,
   });
 
-export const trackEmbedWizardResourceSelected = (
-  resourceId: number,
+export const trackEmbedWizardResourceSelectionCompleted = (
   experience: SdkIframeEmbedSetupExperience,
 ) =>
   trackSimpleEvent({
-    event: "embed_wizard_resource_selected",
-    target_id: resourceId,
+    event: "embed_wizard_resource_selection_completed",
     event_detail: experience,
   });
 
-export const trackEmbedWizardCodeCopied = () =>
-  trackSimpleEvent({
-    event: "embed_wizard_code_copied",
-  });
-
-export const trackEmbedWizardSettingsUpdated = (
+export const trackEmbedWizardOptionsCompleted = (
   settings: Partial<SdkIframeEmbedSettings>,
 ) => {
-  const filteredSettings = filterEmptySettings(settings);
+  const options: Record<string, string> = {
+    theme: settings.theme?.colors ? "custom" : "default",
+    auth: settings.useExistingUserSession ? "user_session" : "sso",
+  };
 
-  const settingKeys = Object.keys(
-    filteredSettings,
-  ) as SdkIframeEmbedSettingKey[];
-
-  const untrackedKeys: SdkIframeEmbedSettingKey[] = [
-    // These keys are not tracked as user settings.
-    "instanceUrl",
+  for (const optionKey of EMBED_OPTIONS_TO_TRACK) {
+    if (!(optionKey in settings)) {
+      continue;
+    }
 
-    // These are tracked via trackEmbedWizardExperienceSelected and trackEmbedWizardResourceSelected
-    "questionId",
-    "dashboardId",
-  ];
+    const value = (settings as any)[optionKey];
 
-  settingKeys.forEach((settingKey) => {
-    if (untrackedKeys.includes(settingKey)) {
-      return;
+    if (value === null || value === undefined) {
+      continue;
     }
 
-    if (settingKey === "useExistingUserSession") {
-      trackSimpleEvent({
-        event: "embed_wizard_auth_selected",
-        event_detail: settings.useExistingUserSession ? "user-session" : "sso",
-      });
+    options[optionKey] = value.toString();
+  }
 
-      return;
-    }
+  const eventDetail = Object.entries(options)
```

Comment:
Not sure we see the same picture or not.
> send as detail if the custom options, custom auth were selected or not

I read it as if we what we send in the detail `default` or `custom` this applies to other step as well.

With all the option like this, I'm not sure it's queryable on stats. If it's not then it has no use.

Reply:
Updated in Slack thread: https://metaboat.slack.com/archives/C063Q3F1HPF/p1759226434274749?thread_ts=1759221821.162359&cid=C063Q3F1HPF

Reply:
Fixed in https://github.com/metabase/metabase/pull/63973/commits/51066ee1023f8a8a5afb1881b51ad1932eaf363c